Carbon.Redis/Abstractions/ICarbonCacheExtensions.cs

```diff
@@ -9,6 +9,24 @@ namespace Carbon.Caching.Abstractions
 {
     public static class ICarbonCacheExtensions
     {
+        private static Serialization SerializationType { get; set; }
+
+        public static void SetSerializationType(Serialization PreferredSerializationType)
+        {
+            if (PreferredSerializationType == Serialization.BinaryFormatter)
```

Comment:
I guess you don't like using switch/case :D (this comment is intended solely as a joke, do not change)

Reply:
If else ftw :p

If you have bunch of conditions, switch case will work like a charm, as it works o(1), comparing to if-else o(n) which makes you right about it. 

However, this method will only have no more than 3 serialization option and method will be called once in a life time.

So it is not a joke, but ignorable in this case